fastparquet/converted_types.py

```diff
@@ -109,15 +128,33 @@ def convert(data, se):
     elif ctype == parquet_thrift.ConvertedType.UINT_64:
         return data.astype(np.uint64)
     elif ctype == parquet_thrift.ConvertedType.JSON:
-        return data.astype('O').str.decode('utf8').map(json.loads)
+        if isinstance(data, list) or data.dtype != "O":
```

Comment:
This is for a set of categories, which must be encoded as if it were an object column; but the conversion here could equally well apply to the fixed string case, so I don't want an asarray(data, dtype="O").
I have some POC work on reading list and map types, not ready to be published anywhere (and maybe not worth polishing in the near-term).